src/upload-button.tsx

```diff
@@ -0,0 +1,356 @@
+    const Dialogs = useDialogs();
+    const [applyToAll, setApplyToAll] = React.useState<boolean>(false);
+
+    const handleReplace = () => {
+        Dialogs.close({ replace: true, applyToAll });
+    };
+
+    const handleSkip = () => {
+        Dialogs.close({ skip: true, applyToAll });
+    };
+
+    const handleCancel = () => {
+        Dialogs.reject(new Error("cancelled"));
+    };
+
+    return (
+        <Modal
+          position="top"
+          // @ts-expect-error incorrect PatternFly typing https://github.com/patternfly/patternfly-react/issues/10361
+          title={fmt_to_fragments(_("Replace file $0?"), <b>{uploadFile.name}</b>)}
+          titleIconVariant="warning"
+          variant={ModalVariant.medium}
+          onClose={handleCancel}
+          isOpen
+          footer={
+              <>
+                  <Button variant="warning" onClick={handleReplace}>{_("Replace")}</Button>
+                  <Button variant="secondary" onClick={handleSkip}>{_("Keep original")}</Button>
+                  <Button variant="link" onClick={handleCancel}>{_("Cancel")}</Button>
+              </>
+          }
+        >
+            <p>
+                {cockpit.format(
+                    _("A file with the same name already exists in \"$0\". Replacing it will overwrite its content."),
+                    path.join('/')
+                )}
+            </p>
+            <Flex
+              spaceItems={{ default: "spaceItems3xl" }}
+              className="conflict-modal-files"
+            >
+                <FlexItem>
+                    <b>{_("New file")}</b>
+                    <p>{cockpit.format_bytes(uploadFile.size)}</p>
+                    <p>{timeformat.dateTime(uploadFile.lastModified)}</p>
+                </FlexItem>
+                <FlexItem>
+                    <b>{_("Original file on server")}</b>
+                    <p>{cockpit.format_bytes(file.size)}</p>
+                    {file.mtime &&
+                    <p>{timeformat.dateTime(file.mtime * 1000)}</p>}
+                </FlexItem>
+            </Flex>
+            {showApplyAll &&
+            <Checkbox
+              id="replace-all"
+              label={_("Apply this action to all conflicting files")}
+              isChecked={applyToAll}
+              onChange={() => setApplyToAll(!applyToAll)}
+            />}
+        </Modal>
+    );
+};
+
+export const UploadButton = ({
+    files,
+    path,
+} : {
+    files: FolderFileInfo[],
+    path: string[],
+}) => {
+    const ref = useRef<HTMLInputElement>(null);
+    const { addAlert } = useContext(FilesContext);
+    const Dialogs = useDialogs();
+    const [showPopover, setPopover] = React.useState(false);
+    const [uploadedFiles, setUploadedFiles] = useState<{[name: string]:
+                                                        {file: File, progress: number, cancel:() => void}}>({});
+
+    const handleClick = () => {
+        if (ref.current) {
+            ref.current.click();
+        }
```

Comment:
Completely optional: can be shortened to `handleClick = ref.current?.click` and thus the helper function eliminated altogether.